src/Units/DoRUnits.java

```diff
@@ -0,0 +1,791 @@
+          moveType, actions, weapons, STAR_VALUE);
+      holdingCapacity = 2;
+      holdables = new Vector<ChassisEnum>(Arrays.asList(ChassisEnum.AIR_LOW));
+    }
+  }
+
+  public static class SubModel extends DoRUnitModel
+  {
+    private static final long serialVersionUID = 1L;
+    private static final int UNIT_COST = 20000;
+    private static final double STAR_VALUE = 1.8;
+    private static final int MAX_FUEL = 70;
+    private static final int IDLE_FUEL_BURN = 1;
+    private static final int MAX_AMMO = 6;
+    private static final int VISION_RANGE = 5;
+    private static final int MOVE_POWER = 6;
+
+    private static final MoveType moveType = new FloatHeavy();
+    private static final UnitActionType[] actions = UnitActionType.COMBAT_VEHICLE_ACTIONS;
+    private static final WeaponModel[] weapons = { new DoRWeapons.SubTorpedoes() };
+
+    public SubModel()
+    {
+      super("Sub", DoRUnitEnum.SUB, UnitRoleEnum.ASSAULT, ChassisEnum.SHIP, UNIT_COST, MAX_AMMO, MAX_FUEL, IDLE_FUEL_BURN, VISION_RANGE, MOVE_POWER, moveType,
+          actions, weapons, STAR_VALUE);
+    }
+  }
+
+  public static class SubSubModel extends SubModel
+  {
+    private static final long serialVersionUID = 1L;
+    private static final int IDLE_FUEL_BURN = 5;
+
+    public SubSubModel()
+    {
+      super();
+      type = DoRUnitEnum.SUB_SUB;
+      chassis = ChassisEnum.SUBMERGED;
+      idleFuelBurn = IDLE_FUEL_BURN;
+      hidden = true;
+    }
+  }
+
+  public static class CarrierModel extends DoRUnitModel
+  {
+    private static final long serialVersionUID = 1L;
+    private static final int UNIT_COST = 28000;
+    private static final double STAR_VALUE = 2.2;
+    private static final int MAX_FUEL = 99;
+    private static final int IDLE_FUEL_BURN = 1;
+    private static final int MAX_AMMO = -1;
+    private static final int VISION_RANGE = 4;
+    private static final int MOVE_POWER = 5;
+
+    private static final MoveType moveType = new FloatHeavy();
+    private static final UnitActionType[] actions = UnitActionType.COMBAT_TRANSPORT_ACTIONS;
+    private static final WeaponModel[] weapons = { new DoRWeapons.CarrierMGun() };
+
+    public CarrierModel() // TODO: Launch.
+    {
+      super("Carrier", DoRUnitEnum.CARRIER, UnitRoleEnum.TRANSPORT, ChassisEnum.SHIP, UNIT_COST, MAX_AMMO, MAX_FUEL, IDLE_FUEL_BURN, VISION_RANGE, MOVE_POWER,
+          moveType, actions, weapons, STAR_VALUE);
+      maxMaterials = 4;
+      holdingCapacity = 2;
+      holdables = new Vector<ChassisEnum>(Arrays.asList(ChassisEnum.AIR_LOW, ChassisEnum.AIR_HIGH));
+    }
+
+    /** DoR Carriers re-supply and repair their cargo at the beginning of every turn. Make it so. */
+    @Override
+    public GameEventQueue getTurnInitEvents(Unit self, MapMaster map)
+    {
+      GameEventQueue events = new GameEventQueue();
+      for( Unit cargo : self.heldUnits )
+        if( !cargo.isFullySupplied() )
```

Comment:
We only heal onboard units if they are also in need of resupply?

Reply:
Good point. Should fix.